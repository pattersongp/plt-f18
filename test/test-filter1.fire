func bool modulo = (int x) => {
    int rem = x - (2 * (x/2));
    if (rem == 0) { return true; }
    else { return false; }
}
 
func bool f = (int x) => {
    if (modulo(x)) { return true; }
    else { return false; }
}

func str p = (str x) => {
    sprint(x);
    return x;
}

func int p1 = (int x) => {
    print(x);
    return x;
}

func int main = () => {
    array[int, int]a;
    int i = 0;
    while (i < 999) {
        a[i] = i;
        i = i+1;
    } 
    sprint("Length of array is: ");
    print(len(a));
    filter(a, modulo);
    sprint("Length of array after filter for even number is: ");
    print(len(a));
    return 0;
}

