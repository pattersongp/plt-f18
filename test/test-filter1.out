Length of array is:
999
Length of array after filter for even number is:
500